### PR TITLE
Refact register knot

### PR DESCRIPTION
**Pull request by luisfelipemisil**
- devEUI is used only to map the device in the integration context, then is generated a new KNoT id for this device and a token is requested to KNoT Cloud.
- Save the device context on a Volume Docker

### CODE REVIEW

File: internal/integration/knot/knot_protocol.go [hunk 1]
```diff
@@ -71,21 +75,18 @@ func newProtocol(conf config.IntegrationKNoTConfig, deviceChan chan entities.Dev
...
+
 	go registerDevices(p.devices, deviceChan)
 	go handlerKnotAMQP(msgChan, deviceChan)
-	go dataControl(deviceChan, p)
+	go knotStateMachineHandler(deviceChan, p)
 
 	return p, nil
 }
 
 //Teste Device
```
Comment (janael-pinheiro) [inline internal/integration/knot/knot_protocol.go]: Teste -> Test

File: internal/integration/knot/knot_protocol.go [hunk 2]
```diff
@@ -94,18 +95,20 @@ func registerDevices(devices map[string]entities.Device, deviceChan chan entitie
...
+		}
 	}
 }
 
 //Load the device's configuration
 func (p *protocol) LoadDeviceOldContext() {
-	if ok := checkFile("/tmp/device/deviceConfig.yaml"); ok {
-		p.readDeviceFile("/tmp/device/deviceConfig.yaml")
-		log.WithFields(log.Fields{"integration": "ConfigFile"}).Info("existe")
+	if ok := checkFile("/etc/knot/deviceContex.yaml"); ok {
```
Comment (janael-pinheiro) [inline internal/integration/knot/knot_protocol.go]: What do you think of assigning the filepath to a constant?

File: internal/integration/knot/knot_protocol.go [hunk 3]
```diff
@@ -179,10 +184,12 @@ func (p *protocol) checkData(device entities.Device) error {
 // Check for device configuration
 func (p *protocol) checkDeviceConfiguration(device entities.Device) error {
 	var ok bool
-	id_pass := 0
+	id_pass := -1
 	// Check if the ids are correct, no repetition
 	for _, data := range device.Config {
-		if data.SensorID != id_pass {
+		if data.Schema.ValueType < 1 || data.Schema.ValueType > 5 {
```
Comment (janael-pinheiro) [inline internal/integration/knot/knot_protocol.go]: What do you think about moving this condition to a function that takes the data.schema.ValueType and returns a boolean? For example, 
func isInvalidValueType(valueType int) bool {
   return valueType < 1 || valueType > 5
}

Also, consider assigning the values 1 and 5 to constants.

File: internal/integration/knot/knot_protocol.go [hunk 4]
```diff
@@ -282,23 +294,23 @@ func initTimeout(deviceChan chan entities.Device, device entities.Device) {
...
 func (p *protocol) checkTimeout(device entities.Device) entities.Device {
-	curDevice := p.devices[device.ID]
-	if device.Error == "timeOut" {
-		if (device.State == entities.KnotNew && curDevice.State == entities.KnotWaitReg) ||
-			(device.State == entities.KnotRegistered && curDevice.State == entities.KnotWaitAuth) ||
-			(device.State == entities.KnotAuth && curDevice.State == entities.KnotWaitConfig) ||
-			(device.State == entities.KnotAlreadyReg && curDevice.State == entities.KnotWaitUnreg) {
+	curDevice := p.devices[device.DevEUI]
+	if device.Error == values.ErrorTimeOut {
+		if (device.State == values.KNoTNew && curDevice.State == values.KNoTWaitReg) ||
```
Comment (janael-pinheiro) [inline internal/integration/knot/knot_protocol.go]: Please extract this condition to a function with a name that describes what is being checked and returns a boolean.

**Review by janael-pinheiro — CHANGES_REQUESTED** [review]
Very good code.

File: internal/integration/knot/knot_protocol.go [hunk 5]
```diff
@@ -94,18 +95,20 @@ func registerDevices(devices map[string]entities.Device, deviceChan chan entitie
...
-	if ok := checkFile("/tmp/device/deviceConfig.yaml"); ok {
-		p.readDeviceFile("/tmp/device/deviceConfig.yaml")
-		log.WithFields(log.Fields{"integration": "ConfigFile"}).Info("existe")
+	if ok := checkFile("/etc/knot/deviceContex.yaml"); ok {
+		p.readDeviceFile("/etc/knot/deviceContex.yaml")
+		log.WithFields(log.Fields{"integration": "ConfigFile"}).Info("exists")
 	} else {
-		p.writeDeviceFile("/tmp/device/deviceConfig.yaml")
-		log.WithFields(log.Fields{"integration": "ConfigFile"}).Info("Criando")
+		p.writeDeviceFile("/etc/knot/deviceContex.yaml")
```
Comment (rtorresmatheus) [inline internal/integration/knot/knot_protocol.go]: Same idea Janael said before. Can this be constant?

File: internal/integration/knot/knot_protocol.go [hunk 6]
```diff
@@ -136,22 +139,24 @@ func (p *protocol) readDeviceFile(name string) {
...
+func (p *protocol) writeDeviceFile(pathFile string) {
 	devices := p.devices
+
 	data, err := yaml.Marshal(&devices)
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	err = os.WriteFile(name, data, 0600)
+	err = os.WriteFile(pathFile, data, 0644)
```
Comment (rtorresmatheus) [inline internal/integration/knot/knot_protocol.go]: What is the 0644 means? Can this be constant?

**Pull request merged**